buster/busterbot.py

```diff
@@ -88,13 +95,20 @@ def update_cfg(self, cfg: BusterConfig):
         self.retriever_cfg = cfg.retriever_cfg
         self.completion_cfg = cfg.completion_cfg
         self.prompt_cfg = cfg.prompt_cfg
+        self.tokenizer_cfg = cfg.tokenizer_cfg
 
         # set the unk. embedding
         self.unk_embedding = self.get_embedding(self.unknown_prompt, engine=self.embedding_model)
 
         # update completer and formatter cfg
+        self.tokenizer = tokenizer_factory(self.tokenizer_cfg)
         self.completer = completer_factory(self.completion_cfg)
-        self.prompt_formatter = prompt_formatter_factory(self.prompt_cfg)
+        self.documents_formatter = document_formatter_factory(
+            tokenizer=self.tokenizer,
+            max_tokens=self.retriever_cfg["max_tokens"]
+            # TODO: move max_tokens from retriever_cfg to somewhere more logical
```

Comment:
wait it seems to be in `prompt_cfg` but not `retriever_cfg`, looking at lines 41-49. How is this not crashing?

Reply:
it should be in both, where are you looking?  We have a max_tokens for the total prompt, but also a max_tokens for the sum of all documents

Reply:
you are right, i added it